pkg/machine/hyperv/stubber.go

```diff
@@ -0,0 +1,577 @@
+		Rootful:        mc.HostUser.Rootful,
+		VMType:         h.VMType(),
+	}
+	// TODO Should this process be fatal on error; currenty, no error is
+	// returned but an error can occur in the func itself
+	// TODO we do not currently pass "noinfo" (quiet) into the StartVM
+	// func so this is hard set to false
+	machine.LaunchWinProxy(winProxyOpts, false)
+
+	winProxyCallbackFunc := func() error {
+		return machine.StopWinProxy(mc.Name, h.VMType())
+	}
+	callbackFuncs.Add(winProxyCallbackFunc)
+
+	if len(mc.Mounts) != 0 {
+		var (
+			dirs       *define.MachineDirs
+			gvproxyPID int
+		)
+		dirs, err = machine.GetMachineDirs(h.VMType())
+		if err != nil {
+			return err
+		}
+		// GvProxy PID file path is now derived
+		gvproxyPIDFile, err := dirs.RuntimeDir.AppendToNewVMFile("gvproxy.pid", nil)
+		if err != nil {
+			return err
+		}
+		gvproxyPID, err = gvproxyPIDFile.ReadPIDFrom()
+		if err != nil {
+			return err
+		}
+
+		executable, err = os.Executable()
+		if err != nil {
+			return err
+		}
+		// Start the 9p server in the background
+		p9ServerArgs := []string{}
+		if logrus.IsLevelEnabled(logrus.DebugLevel) {
+			p9ServerArgs = append(p9ServerArgs, "--log-level=debug")
+		}
+		p9ServerArgs = append(p9ServerArgs, "machine", "server9p")
+
+		for _, mount := range mc.Mounts {
+			if mount.VSockNumber == nil {
+				return fmt.Errorf("mount %s has not vsock port defined", mount.Source)
+			}
+			p9ServerArgs = append(p9ServerArgs, "--serve", fmt.Sprintf("%s:%s", mount.Source, winio.VsockServiceID(uint32(*mount.VSockNumber)).String()))
+		}
+		p9ServerArgs = append(p9ServerArgs, fmt.Sprintf("%d", gvproxyPID))
+
+		logrus.Debugf("Going to start 9p server using command: %s %v", executable, p9ServerArgs)
+
+		fsCmd := exec.Command(executable, p9ServerArgs...)
+
+		if logrus.IsLevelEnabled(logrus.DebugLevel) {
+			err = logCommandToFile(fsCmd, "podman-machine-server9.log")
+			if err != nil {
+				return err
+			}
+		}
+
+		err = fsCmd.Start()
+		if err == nil {
+			logrus.Infof("Started podman 9p server as PID %d", fsCmd.Process.Pid)
+		}
+
+		// Note: No callback is needed to stop the 9p server, because it will stop when
+		// gvproxy stops
+
+		// Finalize starting shares after we are confident gvproxy is still alive.
+		err = startShares(mc)
+	}
+	return err
+}
+
+func resizeDisk(newSize strongunits.GiB, imagePath *define.VMFile) error {
+	resize := exec.Command("powershell", []string{"-command", fmt.Sprintf("Resize-VHD %s %d", imagePath.GetPath(), newSize.ToBytes())}...)
+	logrus.Debug(resize.Args)
+	resize.Stdout = os.Stdout
+	resize.Stderr = os.Stderr
+	if err := resize.Run(); err != nil {
+		return fmt.Errorf("resizing image: %q", err)
+	}
+	return nil
+}
+
+func addNetworkAndReadySocketsToRegistry(mc *vmconfigs.MachineConfig) error {
+	networkHVSock, err := vsock.NewHVSockRegistryEntry(mc.Name, vsock.Network)
+	if err != nil {
+		return err
+	}
+	//eventHVSocket, err := vsock.NewHVSockRegistryEntry(mc.Name, vsock.Events)
```

Comment:
Should this be removed? If we are using a common representation of a ready socket this entire function can be removed.